include/mbedtls/check_config.h

```diff
@@ -177,11 +177,32 @@
 #endif
 #undef MBEDTLS_HAS_MEMSAN
 
+#if defined(MBEDTLS_CCM_C) && (                                        \
+        !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_CAMELLIA_C) && !defined(MBEDTLS_ARIA_C) )
+#error "MBEDTLS_CCM_C defined, but not all prerequisites"
+#endif
+
+#if defined(MBEDTLS_CCM_C) && !defined(MBEDTLS_CIPHER_C)
```

Comment:
Why is this separated from `#if defined` at lines 180-181?
Basically both could be written as:
`#if defined(MBEDTLS_CCM_C) && !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_CAMELLIA_C) && !defined(MBEDTLS_ARIA_C) && !defined(MBEDTLS_CIPHER_C)`


Same for MBEDTLS_GCM_C and MBEDTLS_CHACHAPOLY_C

Reply:
The code is simpler that way. Fewer nested parentheses.

Note that what you posted is different: CCM requires _at least one_ block cipher, whereas you posted code that makes it require _all_ block ciphers.